cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/RunEnvironment.java

```diff
@@ -107,4 +108,36 @@ public ExecutionPath getExecutionPath() {
     public Map<String, SerializableSessionObject> getSerializableDataMap() {
         return serializableDataMap;
     }
+
+    public RunEnvironment cloneWithoutStacks() {
+        @SuppressWarnings("unchecked")
+        Map<String, ? extends Serializable> systemPropertiesCopy =
+                (Map<String, ? extends Serializable>) SerializationUtils.clone((Serializable) systemProperties);
+
+        RunEnvironment runEnvironmentWithoutStacks = new RunEnvironment(systemPropertiesCopy);
+
+        runEnvironmentWithoutStacks.contextStack = new ContextStack();
+        runEnvironmentWithoutStacks.parentFlowStack = new ParentFlowStack();
+
+        @SuppressWarnings("unchecked")
+        Map<String, Serializable> callArgumentsCopy =
+                (Map<String, Serializable>) SerializationUtils.clone((Serializable) callArguments);
+        runEnvironmentWithoutStacks.callArguments = callArgumentsCopy;
+
+        runEnvironmentWithoutStacks.executionPath =
+                (ExecutionPath) SerializationUtils.clone(executionPath);
+
+        @SuppressWarnings("unchecked")
+        Map<String, SerializableSessionObject> serializableDataMapCopy =
+                (Map<String, SerializableSessionObject>) SerializationUtils.clone((Serializable) serializableDataMap);
+        runEnvironmentWithoutStacks.serializableDataMap = serializableDataMapCopy;
+
+        runEnvironmentWithoutStacks.returnValues =
+                (ReturnValues) SerializationUtils.clone(returnValues);
+
+        runEnvironmentWithoutStacks.nextStepPosition = nextStepPosition;
+
+        return runEnvironmentWithoutStacks;
```

Comment:
this is a bit tricky, since if someone , add/changes RunEnvironment  Object he must update this method (cloneWithoutStacks) otherwise weird stuff will happen... what about reverse approach? 
clone normally and just reset (after that) the ContextStack field.
It's more work (less efficient)  but code wise, might be cleaner....

@orius123  WDYT?